fdbserver/storageserver.actor.cpp

```diff
@@ -5121,12 +5145,15 @@ ACTOR Future<Void> getMappedKeyValuesQ(StorageServer* data, GetMappedKeyValuesRe
 			req.reply.send(none);
 		} else {
 			state int remainingLimitBytes = req.limitBytes;
-
+			// create a temporary byte limit for index fetching ONLY, this should be excessive
+			// because readRange is cheap when reading additional bytes
+			const double fraction = 0.8;
```

Comment:
My assumption would actually be that the primary is a much smaller fraction--maybe closer to 20% than 80%. That's just a guess though. Perhaps this should be a knob?

Reply:
I still think this probably should be a knob, but I suppose the other side would be, if we're assuming that overriding from the primary scan is cheap (because it's a single scan), and we only really are concerned about overreading from the secondary scans (because those are random reads and are potentially larger), then I suppose having this overestimate (as a heuristic) is fine.

Reply:
agree, will make it a Knob.